src/main/java/com/vmware/vipclient/i18n/VIPCfgFactory.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.vmware.vipclient.i18n.util.VIPConfig;
+
+public class VIPCfgFactory {
+    /**
+     * A lazy-loaded singleton VIPCfg using the "initialization-on-demand holder" design pattern
+     */
+    private static class GlobalCfgHolder {
+        static final VIPCfgWrapper globalCfg = new VIPCfgWrapper(new VIPCfg());
+    }
+
+    public static VIPCfgWrapper getGlobalCfg() {
+        return GlobalCfgHolder.globalCfg;
+    }
+
+    /**
+     * @deprecated This method was added for backwards compatibility with deprecated {@link VIPCfg#getInstance() getInstance}.
+     * Use the {@link #getGlobalCfg() getGlobalCfg} method.
+     */
+    public static VIPCfgWrapper getCfg() {
```

Comment:
You can remove this directly because it's new.

Reply:
See comment section. Removing this will cause backwards compatibility issues for code that calls VIPCfg.getInstance. 

In new approach, we do not allow instantiation of VIPCfg instance without product name ( like in VIPCfg.getInstance()). Instead, you should use VIpCfgFactory.getCfg(productName, true), or VIPCfgFactory.initialize(cfgBundeName, true). VIPCfgFactory.getGlobalCfg() will return null if product name is not yet set.

Reply:
This is confused because you created an deprecated method.

Reply:
Who will be confused? Product developers will see clearly in the comment section that this is deprecated so they will not use it. They will also see clearly in the comment section which method to use instead.

Do you have any other proposal on how to keep backwards compatibility for VIPCfg.getInstance?